csrc/mmdeploy/net/rknn/rknn_net.cpp

```diff
@@ -0,0 +1,211 @@
+  int ret = rknn_init(&ctx_, model_ptr, content.size(), 0, NULL);
+  if (ret < 0) {
+    MMDEPLOY_ERROR("Load .rknn failed: {}", config.net);
+    return Status(eInvalidArgument);
+  }
+
+  // Get Model Input Output Info
+  rknn_input_output_num io_num;
+  ret = rknn_query(ctx_, RKNN_QUERY_IN_OUT_NUM, &io_num, sizeof(io_num));
+  if (ret != RKNN_SUCC) {
+    MMDEPLOY_ERROR("rknn_query fail! ret= {}", ret);
+    return Status(eFail);
+  }
+  printf("model input num: %d, output num: %d\n", io_num.n_input, io_num.n_output);
+
+  printf("input tensors:\n");
+  rknn_tensor_attr input_attrs[io_num.n_input];
+  memset(input_attrs, 0, sizeof(input_attrs));
+  for (int i = 0; i < io_num.n_input; i++) {
+    input_attrs[i].index = i;
+    ret = rknn_query(ctx_, RKNN_QUERY_INPUT_ATTR, &(input_attrs[i]), sizeof(rknn_tensor_attr));
+    if (ret != RKNN_SUCC) {
+      MMDEPLOY_ERROR("rknn_query fail! ret= {}", ret);
+      return Status(eFail);
+    }
+    dump_tensor_attr(&(input_attrs[i]));
+  }
+  input_attrs_ = std::vector<rknn_tensor_attr>(input_attrs, input_attrs + io_num.n_input);
+
+  printf("output tensors:\n");
+  rknn_tensor_attr output_attrs[io_num.n_output];
+  memset(output_attrs, 0, sizeof(output_attrs));
+  for (int i = 0; i < io_num.n_output; i++) {
+    output_attrs[i].index = i;
+    ret = rknn_query(ctx_, RKNN_QUERY_OUTPUT_ATTR, &(output_attrs[i]), sizeof(rknn_tensor_attr));
+    if (ret != RKNN_SUCC) {
+      MMDEPLOY_ERROR("rknn_query fail! ret= {}", ret);
+      return Status(eFail);
+    }
+    dump_tensor_attr(&(output_attrs[i]));
+  }
+  output_attrs_ = std::vector<rknn_tensor_attr>(output_attrs, output_attrs + io_num.n_output);
+
+  for (int i = 0; i < io_num.n_input; ++i) {
+    // TODO get real data type instead of hard code
+    input_tensors_.emplace_back(TensorDesc{device_, DataType::kFLOAT, {}, "#" + std::to_string(i)});
+  }
+
+  for (int i = 0; i < io_num.n_output; ++i) {
+    // TODO get real data type instead of hard code
+    output_tensors_.emplace_back(
+        TensorDesc{device_, DataType::kFLOAT, {}, "#" + std::to_string(i)});
+  }
+
+  return success();
+}
+
+Result<void> RKNNNet::ForwardAsync(Event* event) { return Status(eNotSupported); }
+
+Result<void> RKNNNet::Deinit() { return success(); }
+
+Result<Span<Tensor>> RKNNNet::GetInputTensors() { return input_tensors_; }
+
+Result<Span<Tensor>> RKNNNet::GetOutputTensors() { return output_tensors_; }
+
+Result<void> RKNNNet::Reshape(Span<TensorShape> input_shapes) {
+  for (size_t i = 0; i < input_shapes.size(); ++i) {
+    input_tensors_[i].Reshape(input_shapes[i]);
+  }
+  return success();
+}
+
+Result<void> RKNNNet::Forward() {
+  OUTCOME_TRY(stream_.Wait());
+
+  rknn_input inputs[input_tensors_.size()];
+  memset(inputs, 0, input_tensors_.size() * sizeof(rknn_input));
+  for (int i = 0; i < input_tensors_.size(); i++) {
+    inputs[i].index = i;
+    inputs[i].pass_through = 0;
+    inputs[i].type = RKNN_TENSOR_FLOAT32;
+    inputs[i].fmt = RKNN_TENSOR_NHWC;
```

Comment:
What are the transform ops in preprocess? If there are `defaultFormatBundle` or `Image2Tensor`, the data format is `NCHW`

Reply:
This is no `defaultFormatBundle` or `Image2Tensor` for rknn now.